src/PlotlyEditor.js

```diff
@@ -145,6 +145,36 @@ class PlotlyEditor extends Component {
         }
         break;
 
+      case EDITOR_ACTIONS.DELETE_SHAPE:
+        if (isNumeric(payload.shapeIndex)) {
+          if (this.props.beforeDeleteShape) {
+            this.props.beforeDeleteShape(payload);
+          }
+          graphDiv.layout.shapes.splice(payload.shapeIndex, 1);
+          if (this.props.afterDeleteShape) {
+            this.props.afterDeleteShape(payload);
+          }
+          if (this.props.onUpdate) {
+            this.props.onUpdate();
+          }
+        }
+        break;
+
+      case EDITOR_ACTIONS.DELETE_IMAGE:
+        if (isNumeric(payload.imageIndex)) {
```

Comment:
I guess on a later pass we could combine these delete 'annotations' 'shapes' 'images' actions into one

Reply:
yeah, this was a dirty copy-paste job... We'll see if it turns out that these evolve in different directions and if not, we can DRY them up.